app/src/main/kotlin/com/google/samples/apps/nowinandroid/MainActivity.kt

```diff
@@ -87,6 +89,14 @@ class MainActivity : ComponentActivity() {
         val splashScreen = installSplashScreen()
         super.onCreate(savedInstanceState)
 
+
+        // Kill NiA if there are main thread policy violations and log the offending call.
+        if (isDebuggable()) {
+            StrictMode.setThreadPolicy(
```

Comment:
Just curious, why not put this early in the `Application` instead, to catch anything we do during `Application.onCreate`?